### ISSUE TITLE
x86 32bit build explicitly used -march=i586

### ISSUE THREAD

**Issue by Vascom** [sadko4u/lsp-plugins#251]
For 32bit x86 build explicitly used `-march=i586` flag.
At fedora all packages must be build for i686.

Can this behavior be changed somehow?
Now I use this to fix the problem:
```
sed -i 's|march=i586|march=i686|' make/system.mk
```

**Comment by sadko4u** [comment]
You can do this by, for example, applying:
```
make config ARCHITECTURE_CFLAGS="-march=i686 -m32"
```

**Comment by Vascom** [comment]
This is the same as changing by sed. No way to do it more automatically?

**Comment by sadko4u** [comment]
How should the build script detect that you're aiming to build for i686 architecture but not for i586?

**Comment by Vascom** [comment]
i686 can be detected from `uname -m` or `arch` commands. Seems that it is not a problem.

**Comment by sadko4u** [comment]
This is true if we build on an 32-bit machine in a 32-bit environment.
OK, I can add some extra detection into the build scripts.

**Comment by Vascom** [comment]
Most of other applications don't require to manually set build arch.

**Comment by sadko4u** [comment]
Updated makefiles scripts. Will be available i 1.2.2

**Comment by sadko4u** [comment]
Available in 1.2.2!
